test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -0,0 +1,955 @@
+    rescue Fluent::ConfigError => error
+      assert error.message.include? 'Unable to obtain metadata parameters:'
+      assert error.message.include? 'vm_id'
+      exception_count += 1
+    end
+    assert_equal 1, exception_count
+  end
+
+  def test_configure_invalid_metadata_missing_all_no_metadata_service_grpc
+    setup_no_metadata_service_stubs
+    Fluent::GoogleCloudOutput::CredentialsInfo.stubs(:project_id).returns(nil)
+    exception_count = 0
+    begin
+      create_grpc_driver(USE_GRPC_CONFIG + CONFIG_MISSING_METADATA_ALL)
+    rescue Fluent::ConfigError => error
+      assert error.message.include? 'Unable to obtain metadata parameters:'
+      assert error.message.include? 'project_id'
+      assert error.message.include? 'zone'
+      assert error.message.include? 'vm_id'
+      exception_count += 1
+    end
+    assert_equal 1, exception_count
+  end
+
+  def test_metadata_loading_grpc
+    setup_gce_metadata_stubs
+    d = create_grpc_driver
+    d.run
+    assert_equal PROJECT_ID, d.instance.project_id
+    assert_equal ZONE, d.instance.zone
+    assert_equal VM_ID, d.instance.vm_id
+    assert_equal false, d.instance.running_on_managed_vm
+  end
+
+  def test_managed_vm_metadata_loading_grpc
+    setup_gce_metadata_stubs
+    setup_managed_vm_metadata_stubs
+    d = create_grpc_driver
+    d.run
+    assert_equal PROJECT_ID, d.instance.project_id
+    assert_equal ZONE, d.instance.zone
+    assert_equal VM_ID, d.instance.vm_id
+    assert_equal true, d.instance.running_on_managed_vm
+    assert_equal MANAGED_VM_BACKEND_NAME, d.instance.gae_backend_name
+    assert_equal MANAGED_VM_BACKEND_VERSION, d.instance.gae_backend_version
+  end
+
+  def test_gce_metadata_does_not_load_when_use_metadata_service_is_false_grpc
+    Fluent::GoogleCloudOutput.any_instance.expects(:fetch_metadata).never
+    d = create_grpc_driver(USE_GRPC_CONFIG + NO_METADATA_SERVICE_CONFIG +
+                           CUSTOM_METADATA_CONFIG)
+    d.run
+    assert_equal CUSTOM_PROJECT_ID, d.instance.project_id
+    assert_equal CUSTOM_ZONE, d.instance.zone
+    assert_equal CUSTOM_VM_ID, d.instance.vm_id
+    assert_equal false, d.instance.running_on_managed_vm
+  end
+
+  def test_gce_used_when_detect_subservice_is_false_grpc
+    setup_gce_metadata_stubs
+    # This would cause the service to be container.googleapis.com if not for the
+    # detect_subservice=false config.
+    setup_container_metadata_stubs
+    d = create_grpc_driver(USE_GRPC_CONFIG + NO_DETECT_SUBSERVICE_CONFIG)
+    d.run
+    assert_equal COMPUTE_SERVICE_NAME, d.instance.service_name
+  end
+
+  def test_grpc_used_when_use_grpc_is_true_grpc
+    setup_gce_metadata_stubs
+    d = create_grpc_driver
+    assert_true d.instance.instance_variable_get(:@use_grpc)
+  end
+
+  def test_grpc_not_used_when_use_grpc_is_not_specified_grpc
+    setup_gce_metadata_stubs
+    d = create_driver
+    assert_false d.instance.instance_variable_get(:@use_grpc)
+  end
```

Comment:
Similar situation as earlier comment. Could be bundled into one method which accepts a driver and a assertion Boolean?